### ISSUE TITLE
Can't map IEnumerable<int> to List<int>

### ISSUE THREAD

**Issue by upta** [AutoMapper/AutoMapper#2387]
Hi there, I'm working on upgrading some old code that used 4.2.0 and am now getting an error when trying to map an IEnumerable<int> to a List<int>

Am I doing something stupid or were their changes that would break this since this mapping was written? :)

Thanks!

### Mapping configuration

```csharp
CreateMap<Handout,ToolHandoutDto>()
    .ForMember( a => a.LogIds,
                a => a.MapFrom( b => b.Logs.Select( c => c.LogId ) ) );
```

### Version: x.y.z
6.1.1

### Expected behavior

Used to map without complaint in my project using 4.2.0

### Actual behavior

Unable to create a map expression from Log.LogId (System.Collections.Generic.IEnumerable`1[System.Int32]) to ToolHandoutDto.LogIds (System.Collections.Generic.List`1[System.Int32])

**Comment by lbargaoanu** [comment]
A repro would help. Make a [gist](https://gist.github.com/lbargaoanu/9c7233441c3a3413cc2b9b9ebb5964a9) that we can execute and see fail.

**Comment by upta** [comment]
I went to make a repro, but it worked fine in a basic POCO scenario; whereas my actual scenario involves EF, QueryableExtentions, etc which is a lot harder to mock together.

I did find that if I change my dto's properties from List<> to IEnumerable<> the problem goes away.  So, in practice I'll just do that, 😁 

But here's the full exception I was getting if that's in anyway helpful to you (probably not, but figured I'd include it anyway ;)  Thanks!

AutoMapper.AutoMapperMappingException: Unable to create a map expression from Log.LogId (System.Collections.Generic.IEnumerable`1[System.Int32]) to ToolHandoutDto.LogIds (System.Collections.Generic.List`1[System.Int32])\r\n\r\nMapping types:\r\nHandout -> ToolHandoutDto\r\nProspect.DomainClasses.Handout -> Prospect.AspNetCore.ToolHandoutDto\r\n\r\nType Map configuration:\r\nHandout -> ToolHandoutDto\r\nProspect.DomainClasses.Handout -> Prospect.AspNetCore.ToolHandoutDto\r\n\r\nProperty:\r\nLogIds\r\n   at AutoMapper.QueryableExtensions.ExpressionBuilder.<CreateMemberBindings>g__CreateMemberBinding15_1(PropertyMap propertyMap, <>c__DisplayClass15_0& )\r\n   at AutoMapper.QueryableExtensions.ExpressionBuilder.CreateMemberBindings(ExpressionRequest request, TypeMap typeMap, Expression instanceParameter, IDictionary`2 typePairCount, LetPropertyMaps letPropertyMaps)\r\n   at AutoMapper.QueryableExtensions.ExpressionBuilder.CreateMapExpressionCore(ExpressionRequest request, Expression instanceParameter, IDictionary`2 typePairCount, TypeMap typeMap, LetPropertyMaps letPropertyMaps)\r\n   at AutoMapper.QueryableExtensions.ExpressionBuilder.CreateMapExpressionCore(ExpressionRequest request, Expression instanceParameter, IDictionary`2 typePairCount, LetPropertyMaps letPropertyMaps, TypeMap& typeMap)\r\n   at AutoMapper.QueryableExtensions.ExpressionBuilder.CreateMapExpression(ExpressionRequest request, IDictionary`2 typePairCount, LetPropertyMaps letPropertyMaps)\r\n   at AutoMapper.QueryableExtensions.Impl.EnumerableExpressionBinder.BindEnumerableExpression(IConfigurationProvider configuration, PropertyMap propertyMap, ExpressionRequest request, ExpressionResolutionResult result, IDictionary`2 typePairCount, LetPropertyMaps letPropertyMaps)\r\n   at AutoMapper.QueryableExtensions.Impl.EnumerableExpressionBinder.Build(IConfigurationProvider configuration, PropertyMap propertyMap, TypeMap propertyTypeMap, ExpressionRequest request, ExpressionResolutionResult result, IDictionary`2 typePairCount, LetPropertyMaps letPropertyMaps)\r\n   at AutoMapper.QueryableExtensions.ExpressionBuilder.<CreateMemberBindings>g__CreateMemberBinding15_1(PropertyMap propertyMap, <>c__DisplayClass15_0& )\r\n   at AutoMapper.QueryableExtensions.ExpressionBuilder.CreateMemberBindings(ExpressionRequest request, TypeMap typeMap, Expression instanceParameter, IDictionary`2 typePairCount, LetPropertyMaps letPropertyMaps)\r\n   at AutoMapper.QueryableExtensions.ExpressionBuilder.CreateMapExpressionCore(ExpressionRequest request, Expression instanceParameter, IDictionary`2 typePairCount, TypeMap typeMap, LetPropertyMaps letPropertyMaps)\r\n   at AutoMapper.QueryableExtensions.ExpressionBuilder.CreateMapExpressionCore(ExpressionRequest request, Expression instanceParameter, IDictionary`2 typePairCount, LetPropertyMaps letPropertyMaps, TypeMap& typeMap)\r\n   at AutoMapper.QueryableExtensions.ExpressionBuilder.CreateMapExpression(ExpressionRequest request, IDictionary`2 typePairCount, LetPropertyMaps letPropertyMaps)\r\n   at AutoMapper.QueryableExtensions.ExpressionBuilder.CreateMapExpression(ExpressionRequest request)\r\n   at AutoMapper.LockingConcurrentDictionary`2.<>c__DisplayClass2_1.<.ctor>b__1()\r\n   at System.Lazy`1.CreateValue()\r\n   at System.Lazy`1.LazyInitValue()\r\n   at System.Lazy`1.get_Value()\r\n   at AutoMapper.QueryableExtensions.ExpressionBuilder.GetMapExpression(Type sourceType, Type destinationType, IDictionary`2 parameters, MemberInfo[] membersToExpand)\r\n   at AutoMapper.QueryableExtensions.ProjectionExpression.To[TResult](IDictionary`2 parameters, IEnumerable`1 memberPathsToExpand)\r\n   at AutoMapper.QueryableExtensions.ProjectionExpression.To[TResult](Object parameters, Expression`1[] membersToExpand)\r\n   at AutoMapper.QueryableExtensions.Extensions.ProjectTo[TDestination](IQueryable source, IConfigurationProvider configuration, Object parameters, Expression`1[] membersToExpand)\r\n   at AutoMapper.QueryableExtensions.Extensions.ProjectTo[TDestination](IQueryable source, Expression`1[] membersToExpand)\r\n   at async Prospect.AspNetCore.DefaultController.Get(?)\r\n   at lambda_method(Closure , Object )\r\n   at async Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeActionMethodAsync(?)\r\n   at async Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeNextActionFilterAsync(?)\r\n   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)\r\n   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)\r\n   at async Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeInnerFilterAsync(?)\r\n   at async Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeNextExceptionFilterAsync(?)
